### DISCUSSION TITLE
Blacklisted column ignored

### DISCUSSION THREAD

**Discussion by sami-sweng** [aarondl/sqlboiler#1064] [General]
If you're having a generation problem please answer these questions before submitting your issue. Thanks!

### What version of SQLBoiler are you using (`sqlboiler --version`)?
4.6.0

### What is your database and version (eg. Postgresql 10)
Cockroach 20.2.4

### If this happened at generation time what was the full SQLBoiler command you used to generate your models? (if not applicable leave blank)
`sqlboiler crdb --config configs/dbal/sqlboiler.toml --wipe --no-hooks --no-rows-affected --no-tests --add-soft-deletes`

### Please provide a relevant database schema so we can replicate your issue (Provided you are comfortable sharing this)
```
CREATE TABLE IF NOT EXISTS table1
(
	id UUID PRIMARY KEY DEFAULT gen_random_uuid(),
	col1 STRING,
	column_to_ignore STRING
);
```

### Further information. What did you do, what did you expect?
In sqlboiler.toml I have

```
output   = "internal/dbal"
pkgname  = "dbal"
wipe     = true

[crdb]
user="root"
pass=""
host="127.0.0.1"
port=26257
dbname="db_name"
sslmode="disable"
whitelist = [
    "table1"
]
blacklist = [
    "table1.column_to_ignore" # not taken into account ?
]
```

But the generator seems to be ignoring the blacklist and still creates a struct containing it. Is there some issue with the way the blacklist is declared?

**Comment by stephenafamo** [comment]
1. Can you share a minimal example of the DB schema so we can test (and maybe reproduce) this?
2. Can you maybe upgrade to the most recent version if that would solve the issue?

**Reply by sami-sweng** [reply]
~Updating to 4.8.3 fixes the issue. Thank you for the support!~

My bad, the update doesn't change the issue.

**Reply by sami-sweng** [reply]
Updated with a minimal schema that displays the issue

**Comment by stephenafamo** [comment]
So, the issue seems to be that you have whitelist defined..

You are meant to use **either** a whitelist **OR** a blacklist, and since the presence of a whitelist is checked first, that makes having a blacklist completely redundant.

**Reply by sami-sweng** [reply]
I have dynamically created tables that should not be included, so it's not possible to only use a blacklist. 
Is there another way to ignore a column given this constraint?

Thank you

**Reply by stephenafamo** [reply]
None at the moment. 

Would it be possible to only whitelist the tables/columns you use?

**Reply by sami-sweng** [reply]
Ah interesting, so I could only whitelist column that I want from table1?

I'll try this ASAP and let you know.

Thank you

**Reply by stephenafamo** [reply]
Also, wildcards in whitelists/blacklist will be released in the next version, so that may make your configuration easier.
